src/main/java/com/google/gcp_variant_transforms/library/VariantToBqUtilsImpl.java

```diff
@@ -0,0 +1,254 @@
+import htsjdk.variant.vcf.VCFInfoHeaderLine;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Implementation class for {@link VariantToBqUtils}. It provides functionalities to set default
+ * values and convert values to the types defined by VCF Header.
+ */
+public class VariantToBqUtilsImpl implements VariantToBqUtils, Serializable {
+  public String getReferenceBases(VariantContext variantContext) {
+    Allele referenceAllele = variantContext.getReference();
+    String referenceBase = referenceAllele.getDisplayString();
+    return replaceMissingWithNull(referenceBase);
+  }
+
+  public List<String> getNames(VariantContext variantContext) {
+    String names = variantContext.getID();
+    String[] splittedNamesBySemiColon = names.split(";");
+    List<String> nameList = new ArrayList<>();
+    for (String name : splittedNamesBySemiColon) {
+      nameList.add(replaceMissingWithNull(name));
+    }
+    return nameList;
+  }
+
+  public List<TableRow> getAlternateBases(VariantContext variantContext) {
+    List<TableRow> altMetadata = new ArrayList<>();
+    List<Allele> altAlleles = variantContext.getAlternateAlleles();
+    // If field value is '.', alternateAllelesList will be empty.
+    if (altAlleles.isEmpty()) {
+      TableRow subRow = new TableRow();
+      subRow.set(Constants.ColumnKeyConstants.ALTERNATE_BASES_ALT, null);
+      altMetadata.add(subRow);
+    } else {
+      for (Allele altAllele : altAlleles) {
+        TableRow subRow = new TableRow();
+        String altBase = altAllele.getDisplayString();
+        subRow.set(Constants.ColumnKeyConstants.ALTERNATE_BASES_ALT, altBase);
+        altMetadata.add(subRow);
+      }
+    }
+    return altMetadata;
+  }
+
+  public Set<String> getFilters(VariantContext variantContext) {
+    Set<String> filters = variantContext.getFiltersMaybeNull();
+    if (filters != null && filters.isEmpty()) {
+      // If the filter is an empty set, it refers to "PASS".
+      filters = new HashSet<>();
+      filters.add("PASS");
+    }
+    return filters;
+  }
+
+  public void addInfo(TableRow row, VariantContext variantContext,
+                      List<TableRow> altMetadata, VCFHeader vcfHeader, int expectedAltCount) {
+    // Iterate all Info field in VCFHeader, if current record does not have the field, skip it.
+    for (VCFInfoHeaderLine infoHeaderLine : vcfHeader.getInfoHeaderLines()) {
+      String attrName = infoHeaderLine.getID();
+      if (variantContext.hasAttribute(attrName)) {
+        Object value = variantContext.getAttribute(attrName);
+        VCFInfoHeaderLine infoMetadata = vcfHeader.getInfoHeaderLine(attrName);
+        VCFHeaderLineType infoType = infoMetadata.getType();
+        VCFHeaderLineCount infoCountType = infoMetadata.getCountType();
+        if (infoCountType == VCFHeaderLineCount.A || infoCountType == VCFHeaderLineCount.R) {
```

Comment:
This is redundant || statement with sub check, just have one if for A, and one else if for R.